src/agent/main.cpp

```diff
@@ -109,7 +108,7 @@ static const struct option  kOptions[]   = {
 
 static void HandleSignal(int aSignal)
 {
-    sShouldTerminate = true;
+    MainloopManager::GetInstance().BreakMainloop();
```

Comment:
Is it safe to call GetInstance() in another thread?

Reply:
Yes, static variable initialization is Thread-safe.